l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -134,124 +129,150 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
+    function setAssetAddress(bytes32 _additionalData, address _assetAddress) external {
+        address sender = msg.sender == address(nativeTokenVault) ? NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS : msg.sender;
+        bytes32 assetInfo = keccak256(abi.encode(uint256(block.chainid), sender, _additionalData)); /// todo make other asse
+        assetAddress[assetInfo] = _assetAddress;
+        emit AssetRegistered(assetInfo, _assetAddress, _additionalData, sender);
+    }
+
     /// @notice Allows bridgehub to acquire mintValue for L1->L2 transactions.
     /// @dev If the corresponding L2 transaction fails, refunds are issued to a refund recipient on L2.
     function bridgehubDepositBaseToken(
         uint256 _chainId,
+        bytes32 _assetInfo,
         address _prevMsgSender,
-        address _l1Token,
         uint256 _amount
     ) external payable virtual onlyBridgehubOrEra(_chainId) whenNotPaused {
-        if (_l1Token == ETH_TOKEN_ADDRESS) {
-            require(msg.value == _amount, "L1SharedBridge: msg.value not equal to amount");
-        } else {
-            // The Bridgehub also checks this, but we want to be sure
-            require(msg.value == 0, "ShB m.v > 0 b d.it");
-
-            uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
-            require(amount == _amount, "3T"); // The token has non-standard transfer logic
-        }
+        (address l1Asset, bytes32 assetInfo) = _getAssetProperties(_assetInfo);
+        _transferAllowanceToNTV(assetInfo, _amount, _prevMsgSender);
+        /* solhint-disable no-unused-vars */
+        bytes memory bridgeMintData = IL1StandardAsset(l1Asset).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: 0,
+            _assetInfo: assetInfo,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
 
-        if (!hyperbridgingEnabled[_chainId]) {
-            chainBalance[_chainId][_l1Token] += _amount;
-        }
         // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
-        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _l1Token, _amount);
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetInfo, _amount);
+        // We could use this data instead of the above
+        // emit BridgehubMintData(bridgeMintData);
+    }
+
+    /// @dev for backwards compatibility and to automatically register l1 assets, and we return the correct info.
+    function _getAssetProperties(bytes32 _assetInfo) internal returns (address l1Asset, bytes32 assetInfo) {
+        l1Asset = assetAddress[_assetInfo];
+        assetInfo = _assetInfo;
+        if (l1Asset == address(0) && (uint256(_assetInfo) <= type(uint160).max)) {
```

Comment:
please add comments for this logic & why do we treat these tokens in a special manner